protocols/v2/framing-sv2/src/framing2.rs

```diff
@@ -129,30 +134,28 @@ impl<'a, T: Serialize + GetSize, B: AsMut<[u8]> + AsRef<[u8]>> Frame<'a, T> for
         }
     }
 
-    // self can be either serialized (it cointain an AsMut<[u8]> with the serialized data or
-    // deserialized it contain the rust type that represant the Sv2 message. If the type is
-    // deserialized self.paylos.is_some() is true. To get the serialized payload the inner type
-    // should be serialized and this function should never be used, cause is intended as a fast
-    // function that return a reference to an already serialized payload. For that the function
-    // panic.
+    /// `self` can be either serialized (`self.serialized` is `Some()`) or
+    /// deserialized (`self.serialized` is `None`, `self.payload` is `Some()`).
+    /// This function is only intended as a fast way to get a reference to an
+    /// already serialized payload. If the frame has not yet been
+    /// serialized, this function should never be used (it will panic).
```

Comment:
sorry @Fi3 I don't fully understand the suggestion

I did some research but couldn't find an easy way to print a warning during compilation.
I only found [this pre-RFC](https://internals.rust-lang.org/t/pre-rfc-add-compile-warning-macro/9370) which proposes a `compile_warning!` macro but it will probably take a long time before this is available.

Do you have some examples on how to achieve this?

Reply:
No I was just supposing that it was already doable on stable, maybe we can leave a comment for our future self.

Reply:
well, as we already discussed briefly, I believe the `framing_sv2` crate should be refactored in the future.

the `Frame` trait is forcing `Sv2Frame` and `HandShakeFrame` to share APIs while they don't really share a lot of common features, and as a consequence we end up with APIs that are inuntuitive and dirty (e.g.: panic on undersired scenarios)

so if we're taking a note for the future, I think we should simply refactor this crate (removing the `Frame` trait) so that the APIs don't need to be emitting compilation warnings in order to avoid footguns

Reply:
I wrote a summary of all things to be refactored on `framing_sv2` in the future here:
https://github.com/stratum-mining/stratum/issues/903